model/generate.go

```diff
@@ -118,57 +117,46 @@ func ParseProjectFromJSON(data []byte) (GeneratedProject, error) {
-func (g *GeneratedProject) NewVersion() (*Project, *ParserProject, *Version, *task.Task, *projectMaps, error) {
-	// Get task, version, and project.
-	t, err := task.FindOneId(g.TaskID)
-	if err != nil {
-		return nil, nil, nil, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusInternalServerError, Message: errors.Wrapf(err, "error finding task %s", g.TaskID).Error()}
-	}
-	if t == nil {
-		return nil, nil, nil, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: fmt.Sprintf("unable to find task %s", g.TaskID)}
-	}
-	v, err := VersionFindOneId(t.Version)
-	if err != nil {
-		return nil, nil, nil, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusInternalServerError, Message: errors.Wrapf(err, "error finding version %s", t.Version).Error()}
-	}
-	if v == nil {
-		return nil, nil, nil, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: fmt.Sprintf("unable to find version %s", t.Version)}
-	}
-	p, pp, err := LoadProjectForVersion(v, t.Project, false)
-	if err != nil {
-		return nil, nil, nil, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: errors.Wrapf(err, "error getting project for version %s", t.Version).Error()}
-	}
-
+func (g *GeneratedProject) NewVersion(p *Project, pp *ParserProject, v *Version) (*Project, *ParserProject, *Version, *projectMaps, error) {
 	// Cache project data in maps for quick lookup
 	cachedProject := cacheProjectData(p)
 
 	// Validate generated project against original project.
-	if err = g.validateGeneratedProject(p, cachedProject); err != nil {
+	if err := g.validateGeneratedProject(p, cachedProject); err != nil {
 		// Return version in this error case for handleError, which checks for a race. We only need to do this in cases where there is a validation check.
-		return nil, pp, v, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: errors.Wrap(err, "generated project is invalid").Error()}
+		return nil, pp, v, nil, errors.Wrap(err, "generated project is invalid")
 	}
 
 	newPP, err := g.addGeneratedProjectToConfig(pp, v.Config, cachedProject)
 	if err != nil {
-		return nil, nil, nil, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: errors.Wrap(err, "error creating config from generated config").Error()}
+		return nil, nil, nil, nil, errors.Wrap(err, "error creating config from generated config")
 	}
 	newPP.Id = v.Id
 	p, err = TranslateProject(newPP)
 	if err != nil {
-		return nil, nil, nil, nil, nil,
-			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: errors.Wrap(err, "error translating project").Error()}
+		return nil, nil, nil, nil, errors.Wrap(err, "error translating project")
 	}
-	return p, newPP, v, t, &cachedProject, nil
+	return p, newPP, v, &cachedProject, nil
 }
 
 func (g *GeneratedProject) Save(ctx context.Context, p *Project, pp *ParserProject, v *Version, t *task.Task, pm *projectMaps) error {
```

Comment:
public function with a private type as an argument is awkward.

Reply:
100% yes, but I don't think refactoring this should be within the scope of this ticket.